### PR TITLE
Pull requests to add integration specs

### PR DESCRIPTION
**Pull request by TechnoAndy**
In this project, I implemented the following:

-  N+1 problem are solved when fetching all posts and their comments for a user
- Screenshots below 
- Before

Screenshot (222)

Screenshot (224)

- After

Screenshot (223)

Screenshot (225)

- Used Capybara to write integration tests for each view in my project.

### CODE REVIEW

File: spec/features/post_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    end
+
+    it 'should display the number of likes' do
+      expect(page).to have_content(@posts.first.likes_counter)
+    end
+
+    it 'should include a section for paginations' do
+      expect(page).to have_content('Pagination')
+    end
+  end
```
Comment (TedLivist) [inline spec/features/post_index_spec.rb:72]: - [ ] When writing tests we want to cover as many scenarios as possible. Kindly add tests for the following scenarios on the Post index page.

  - Tests that assert you can see a post's title.
  - Tests that assert you can see the first comments on a post.
  - Tests that redirect to the post's show page when you click on a post.

File: spec/features/post_show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    it 'should display the post body' do
+      expect(page).to have_content(@posts.first.text)
+    end
+
+    it 'should display the name of the commentor' do
+      @posts.first.comments.each do |comment|
+        expect(page).to have_content(comment.author.name)
+      end
+    end
+  end
```
Comment (TedLivist) [inline spec/features/post_show_spec.rb:55]: - [ ] You are missing tests for the following scenarios in the Post show page.

  - Tests that assert you can see the post's title.
  - Tests that assert you can see the comment each commentor left.

File: spec/features/user_index_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    it 'displays the profile picture of the user' do
+      @user.each do |user|
+        expect(page).to have_css("img[src='#{user.photo}']")
+      end
+    end
+    it 'displays the number of posts for each user' do
+      @user.each do |user|
+        expect(page).to have_content(user.post_counter)
+      end
+    end
```
Comment (TedLivist) [inline spec/features/user_index_spec.rb:38]: - [ ] Kindly add tests for the following scenario:

  - Tests that redirects to the user's show page when you click on a user.

File: spec/features/user_show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      @user.first.recent_posts.each do |post|
+        expect(page).to have_content(post.title)
+      end
+    end
+
+    describe 'GET /index' do
+      it 'has a link to user show page' do
+        expect(page).to have_selector("a[href='#{user_path(@user1.id)}']")
+      end
+    end
```
Comment (TedLivist) [inline spec/features/user_show_spec.rb:48]: - [ ] Does this test belong here? I think it should be grouped with the others.

Comment (TedLivist) [inline spec/features/user_show_spec.rb:48]: - [ ] Kindly implement tests for the following scenarios:

  * Tests that assert you can see the user's bio.
  * Tests that redirect to the post's show page when you click on a user's post.
  * Tests that redirect to the user's post's index page when you click on a **see all posts**.
  -

Comment (TechnoAndy) [reply spec/features/user_show_spec.rb:48]: Change made

File: spec/features/user_index_spec.rb (hunk 3 above)
Comment (TechnoAndy) [reply spec/features/user_index_spec.rb:38]: Change made

Comment (rloterh) [reply spec/features/user_index_spec.rb]: resolved! ✔️

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Andy, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🏃‍♀️ 

## Highlights 👍  
- Good job so far working on integration specs!

## Required Changes ♻️
_Kindly check the inline comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/features/post_index_spec.rb (hunk 1 above)
Comment (rloterh) [reply spec/features/post_index_spec.rb]: Needs attention!

File: spec/features/user_show_spec.rb (hunk 4 above)
Comment (rloterh) [reply spec/features/user_show_spec.rb]: Needs attention!Needs attention!

File: spec/features/post_index_spec.rb (hunk 1 above)
Comment (Whoistolu) [reply spec/features/post_index_spec.rb]: - Well resolved @TechnoAndy

File: spec/features/post_show_spec.rb (hunk 2 above)
Comment (Whoistolu) [reply spec/features/post_show_spec.rb]: - Well resolved 👍🏽

File: spec/features/user_index_spec.rb (hunk 3 above)
Comment (Whoistolu) [reply spec/features/user_index_spec.rb]: - Well resolved 👍🏽

File: spec/features/user_show_spec.rb (hunk 4 above)
Comment (Whoistolu) [reply spec/features/user_show_spec.rb]: - Well resolved 👍🏽

File: spec/features/user_index_spec.rb [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+        expect(page).to have_content("Number of posts: #{user.post_counter}")
+      end
+    end
+
+    describe 'GET /index' do
+      it 'has a link to user show page' do
+        expect(page).to have_selector("a[href='#{user_path(@user1.id)}']")
+      end
+    end
+  end
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply spec/features/user_index_spec.rb]: - Well resolved 👍🏽

File: spec/features/user_show_spec.rb [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+    it "should show a button to see all user's posts" do
+      expect(page).to have_link('See all posts')
+    end
+
+    it "should redirect to the user's posts index page when the 'See all posts' button is clicked" do
+      click_link('See all posts')
+      expect(page).to have_current_path(user_posts_path(@user_one))
+    end
+  end
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply spec/features/user_show_spec.rb]: - Well resolved 👍🏽

File: spec/features/post_index_spec.rb [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+
+    it 'should how many likes a post has' do
+      expect(page).to have_content(@posts.first.likes_counter)
+    end
+
+    it 'should include a section for paginations' do
+      expect(page).to have_content('Pagination')
+    end
+  end
+end
```
[the remark this answers is not in the document]
Comment (Whoistolu) [reply spec/features/post_index_spec.rb]: - Well resolved 👍🏽

File: spec/features/post_show_spec.rb (hunk 2 above)
[the remark this answers is not in the document]
Comment (Whoistolu) [reply spec/features/post_show_spec.rb]: - Well resolved 👍🏽

File: spec/features/user_index_spec.rb (hunk 5 above)
[the remark this answers is not in the document]
Comment (Whoistolu) [reply spec/features/user_index_spec.rb]: - Well resolved 👍🏽

File: spec/features/user_show_spec.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (Whoistolu) [reply spec/features/user_show_spec.rb]: - Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @TechnoAndy 

Great job implementing the recommended changes.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**